### PR TITLE
Replace X links with Mastodon

### PR DESCRIPTION
**Pull request by mgax**
Fixes https://github.com/wagtail/wagtail/issues/12459

---
The Mastodon icon link is broken, probably because we're using the wrong ID value.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,8 +19,8 @@
     <a href="https://pypi.python.org/pypi/wagtail/">
         <img src="https://img.shields.io/pypi/dm/wagtail?logo=Downloads" alt="Monthly downloads" />
     </a>
-    <a href="https://x.com/WagtailCMS">
-        <img src="https://img.shields.io/twitter/follow/WagtailCMS?style=social&logo=twitter" alt="follow on X">
+    <a href="https://fosstodon.org/@wagtail">
+        <img src="https://img.shields.io/mastodon/follow/109316228974672802?domain=https%3A%2F%2Ffosstodon.org&style=social" alt="Follow @wagtail@fosstodon.org">
```
Comment (thibaudcolas) [inline README.md:23]:
```suggestion
        <img src="https://img.shields.io/mastodon/follow/109308882653647818?domain=https%3A%2F%2Ffosstodon.org&style=social" alt="Follow @wagtail@fosstodon.org">
```

**Review by thibaudcolas — APPROVED** [review]
Looking great!

**Comment by thibaudcolas** [comment]
Thank you @mgax, I’ve "backported" this to the upcoming Wagtail 6.3 release so we can add it in the release note for that.

Re the Mastodon account ID, for future ref, I got it from https://fosstodon.org/api/v1/accounts/lookup?acct=wagtail@fosstodon.org

**Comment by mgax** [comment]
Awesome, thanks @thibaudcolas for that link!

**Pull request merged**